# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/telemetry/Histograms.json
# Commit: 00bb17e2a6a7
# Full Hash: 00bb17e2a6a73f9c66e861adae426818ff4d8caa
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-21 21:20:56
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r e98350675c5a -r 00bb17e2a6a7 toolkit/components/telemetry/Histograms.json
--- a/toolkit/components/telemetry/Histograms.json	Thu Sep 21 08:42:26 2023 +0000
+++ b/toolkit/components/telemetry/Histograms.json	Thu Sep 21 08:59:03 2023 +0000
@@ -17736,6 +17736,18 @@
     "kind": "enumerated",
     "description": "The reason why a HTTP transaction is restarted."
   },
+  "HTTP_CONNECTION_CLOSE_REASON": {
+    "record_in_processes": ["main", "socket"],
+    "products": ["firefox"],
+    "alert_emails": ["necko@mozilla.com", "kershaw@mozilla.com"],
+    "expires_in_version": "never",
+    "kind": "enumerated",
+    "keyed": true,
+    "n_values": 50,
+    "bug_numbers": [1784261],
+    "releaseChannelCollection": "opt-out",
+    "description": "Reason for HTTP connection closure, keyed by properties like HTTP version and connection state."
+  },
   "TRANSACTION_ECH_RETRY_WITH_ECH_COUNT": {
     "record_in_processes": ["main", "socket"],
     "products": ["firefox"],
